docker-entrypoint.sh

```diff
@@ -507,10 +504,20 @@ _main() {
 				local COLLATE=unset
 				local CTYPE=unset
 				local ENCODING=unset
-				COLLATE=$(echo 'SHOW LC_COLLATE' | "${OLDPATH}/bin/postgres" --single -D "${OLD}" "${POSTGRES_DB}" | grep 'lc_collate = "' | cut -d '"' -f 2)
-				CTYPE=$(echo 'SHOW LC_CTYPE' | "${OLDPATH}/bin/postgres" --single -D "${OLD}" "${POSTGRES_DB}" | grep 'lc_ctype = "' | cut -d '"' -f 2)
+
 				ENCODING=$(echo 'SHOW SERVER_ENCODING' | "${OLDPATH}/bin/postgres" --single -D "${OLD}" "${POSTGRES_DB}" | grep 'server_encoding = "' | cut -d '"' -f 2)
-				POSTGRES_INITDB_ARGS="--locale=${COLLATE} --lc-collate=${COLLATE} --lc-ctype=${CTYPE} --encoding=${ENCODING}"
+
+				# LC_COLLATE and LC_TYPE have been removed with PG v16
+				# https://www.postgresql.org/docs/release/16.0/
+				if [ "${PGVER}" -lt 16 ]; then
+					COLLATE=$(echo 'SHOW LC_COLLATE' | "${OLDPATH}/bin/postgres" --single -D "${OLD}" "${POSTGRES_DB}" | grep 'lc_collate = "' | cut -d '"' -f 2)
+					CTYPE=$(echo 'SHOW LC_CTYPE' | "${OLDPATH}/bin/postgres" --single -D "${OLD}" "${POSTGRES_DB}" | grep 'lc_ctype = "' | cut -d '"' -f 2)
+
+					POSTGRES_INITDB_ARGS="--locale=${COLLATE} --lc-collate=${COLLATE} --lc-ctype=${CTYPE} --encoding=${ENCODING}"
+				else
+					POSTGRES_INITDB_ARGS="--encoding=${ENCODING}"
+				fi
```

Comment:
not sure if there are any consequences when leaving out `--lc-collate` when booting a Postgres v16 database ...

Reply:
Good catch.  I guess we'll find out, though no-one's raised any issues yet and it's been about a year. :grimacing: